### PR TITLE
Laboratory Work #1, Darya Krinochkina - 20FPL1

### PR DESCRIPTION
**Pull request by DashaKrinochkina**

### CODE REVIEW

File: pipeline.py [hunk 1]
```diff
@@ -92,8 +93,13 @@ def validate_dataset(path_to_validate):
...
 def main():
-    # YOUR CODE HERE
-    pass
+    html = 'https://lingngu.elpub.ru/jour/issue/archive'
+    response = requests.get(html, headers={'user-agent': 'Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 '
+                                                         '(KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36'})
+    page_code = response.text
+
+    with open('page_code.html', 'w', encoding='utf-8') as file:
+        file.write(page_code)
```
Comment (dmitry-uraev) [inline pipeline.py:102]: Fine. Please, continue your work.

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]

File: constants.py [hunk 2]
```diff
@@ -7,3 +7,5 @@
 PROJECT_ROOT = Path(__file__).parent
 ASSETS_PATH = PROJECT_ROOT / 'tmp' / 'articles'
 CRAWLER_CONFIG_PATH = PROJECT_ROOT / 'scrapper_config.json'
+HEADERS = {'user-agent': 'Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) '
```
Comment (dmitry-uraev) [inline constants.py:10]: good

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+requests==2.27.1
```
Comment (dmitry-uraev) [inline requirements.txt:1]: please, add wget as well , as crawler_config checks are failing without it

File: scrapper.py [hunk 4]
```diff
@@ -1,6 +1,18 @@
...
+import re
+import os
+import requests
+from bs4 import BeautifulSoup
+
+from core_utils.article import Article
+from core_utils.pdf_utils import PDFRawFile
+
+from constants import ASSETS_PATH
+# from constants import CRAWLER_CONFIG_PATH ... idk what to do with this
```
Comment (dmitry-uraev) [inline scrapper.py:14]: use CONFIG_PATH to get all scrapper parameters - seed urls, number of articles, headers etc

File: scrapper.py [hunk 5]
```diff
@@ -26,36 +38,131 @@ class Crawler:
...
+        container = pdf_bs.find('div', id='pdfDownloadLinkContainer')
+        download_link = container.find('a')['href']
+
+        pdf = PDFRawFile(download_link, self.article_id)
+
+        pdf.download()
+        self.article.text = pdf.get_text()
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
```
Comment (dmitry-uraev) [inline scrapper.py:112]: what about finding date?

File: scrapper.py [hunk 6]
```diff
@@ -26,36 +38,131 @@ class Crawler:
...
+        raise IncorrectNumberOfArticlesError
+
+    if max_articles > 200:
+        raise NumberOfArticlesOutOfRangeError
+
+    prepare_environment(ASSETS_PATH)
+    return seed_urls, max_articles
 
 
 if __name__ == '__main__':
```
Comment (dmitry-uraev) [inline scrapper.py:168]: it is time to test your scrapper in this block - call it with parameters from config_path (use variable imported) and get all articles

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
Good job. Lets finish all till the last review next Wednesday

File: scrapper.py [hunk 7]
```diff
@@ -26,38 +39,135 @@ class Crawler:
...
+        if not re.match(r'https?://', url):  # https or http
+            raise IncorrectURLError
+
+    if not isinstance(max_articles, int) or max_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if max_articles > 200:
+        raise NumberOfArticlesOutOfRangeError
+
+    prepare_environment(ASSETS_PATH)
```
Comment (demid5111) [inline scrapper.py:161]: this should not happen during the *validation* of config. recall the SRP (single responsibility principle we were talking about during the last lecture?)

File: scrapper.py [hunk 8]
```diff
@@ -26,38 +39,135 @@ class Crawler:
     Crawler implementation
     """
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
```
Comment (demid5111) [inline scrapper.py:43]: you do not use it anywhere. it is a mistake

**Review by demid5111 — CHANGES_REQUESTED** [review]

File: scrapper.py [hunk 9]
```diff
@@ -26,38 +41,143 @@ class Crawler:
...
+
+        article_author = article_bs.find('meta', {'name': 'DC.Creator.PersonalName'})
+        if not article_author:
+            article_author = 'NOT FOUND'
+        else:
+            article_author = article_author['content']
+        self.article.author = article_author
+
+        date_raw = article_bs.find('meta', {'name': 'DC.Date.dateSubmitted'})['content']
+        article_date = datetime.datetime.strptime(date_raw, '%Y-%m-%d')
```
Comment (dmitry-uraev) [inline scrapper.py:135]: what about finding date in a separate function? SRP

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
I saw green PR, but now it is not. Make sure you have green PR on Friday assignment

File: scrapper.py (hunk 9 above)
Comment (DashaKrinochkina) [reply scrapper.py:135]: If you don't mind, I will answer here on your overall comment. Can you tell me why checks do not work anymore? I don't know how to start checks, they don't do it automatically and git hub writes that there are 0 checks. And now my PR don't have any status (green or red)

Comment (DashaKrinochkina) [reply scrapper.py:141]: done

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
@DashaKrinochkina good job, keep going!

File: pipeline.py [hunk 10]
```diff
@@ -72,30 +86,96 @@ class TextProcessingPipeline:
...
+        raise EmptyDirectoryError
+
+    for child in children:
+        if 'raw.txt' in str(child):
+            with open(child, 'r', encoding='utf-8') as file:
+                text = file.read()
+                if not text:
+                    raise InconsistentDatasetError
+
+    counter_t = 0
```
[the remark this answers is not in the document]
Comment (DashaKrinochkina) [reply pipeline.py]: done

**Pull request closed without merging**